### ISSUE TITLE
thingShadow.register in example fails to run callback

### ISSUE THREAD

**Issue by rvillanueva** [aws/aws-iot-device-sdk-js#126]
The options parameter for `thingShadow.register` is not truly optional. For instance, this will work:

```
thingsShadows.on('connect', () => {
  console.log('Connected, registering...');
  thingsShadows.register(shadowId, {}, () => {
    console.log('registered');
  })
})
```

But when you remove the placeholder options object like below, it no longer calls the callback.

```
thingsShadows.on('connect', () => {
  console.log('Connected, registering...');
  thingsShadows.register(shadowId, () => {
    console.log('registered');
  })
})
```

This causes the README thingShadow example to fail.

**Comment by rvillanueva** [comment]
Submitted a pull request #127

**Comment by fengsongAWS** [comment]
Hi @rvillanueva ,
We are aware of this issue and thanks for your interest and pull request. We will have this fix in future release.
Thanks

**Comment by fengsongAWS** [comment]
Fixed in v2.0.0
